### ISSUE TITLE
windows 2016, apache 2.4 error directory

### ISSUE THREAD

**Issue by FRANCOSIBILLINI** [qgis/QGIS#33460]
qgis server installation under windows 2016 STD - APACHE 2.4 LAST EDITION

Following your tutorial - Editing httpd.conf

THIS WAY apache doesn't start:
<Directory "c:/OSGeo4W64/apps/qgis-ltr/bin">
 SetHandler cgi-script
 AllowOverride None
 Options ExecCGI
 Order allow,deny
 Allow from all
 Require all granted
</Directory>

This way APACHE start but qgis server doen't works (error.log: "....End of script output before headers: qgis_mapserv.fcgi.exe" )

<Directory "c:/OSGeo4W64/apps/qgis-ltr/bin">
 SetHandler cgi-script
 AllowOverride None
 Options ExecCGI
 #Order allow,deny
 #Allow from all
 Require all granted
</Directory>

**Comment by gioman** [comment]
> Following your tutorial

@FRANCOSIBILLINI what tutorial?

**Comment by FRANCOSIBILLINI** [comment]
https://docs.qgis.org/testing/en/docs/user_manual/working_with_ogc/server/getting_started.html#installation-on-windows

**Comment by gioman** [comment]
@FRANCOSIBILLINI follow this (at least the first part), using the provided Apache binaries and config files and see if it helps: https://github.com/NaturalGIS/qgis_server_and_lizmap_on_windows

**Comment by FRANCOSIBILLINI** [comment]
Thanks for your suggestion, but I suppose, is it incomplete? 
Anyway...
![Immagine](https://user-images.githubusercontent.com/37626686/71169490-5616f400-2259-11ea-83c5-e348f6f35c30.png)
 
I've installed again 

1) XAMPP (just apache and php)
2) OSGeo4W (just ltr-3.4.14.1-desktop and ltr-3.4.14.1-server + dependencies )
3 update httpd.conf as:
ScriptAlias /cgi-bin/ "C:/OSGeo4W64/apps/qgis-ltr/bin/"

<Directory "C:/OSGeo4W64/apps/qgis-ltr/bin">
 SetHandler cgi-script
 AllowOverride None
 Options ExecCGI
 Order allow,deny
 Allow from all
 Require all granted
</Directory>

AddHandler cgi-script .cgi .pl .asp .exe
 
SetEnv GDAL_DATA "C:\OSGeo4W64\share\gdal"
SetEnv QGIS_AUTH_DB_DIR_PATH "C:\OSGeo4W64\apps\qgis-ltr\resources"
SetEnv PYTHONHOME "C:\OSGeo4W64\apps\Python37"
SetEnv PATH "C:\OSGeo4W64\bin;C:\OSGeo4W64\apps\qgis-ltr\bin;C:\OSGeo4W64\apps\Qt5\bin;C:\WINDOWS\system32;C:\WINDOWS;C:\WINDOWS\System32\wbem"
SetEnv QGIS_PREFIX_PATH "C:\OSGeo4W64\apps\qgis-ltr"
SetEnv QT_PLUGIN_PATH "C:\OSGeo4W64\apps\qgis-ltr\plugins;C:\OSGeo4W64\apps\Qt5\plugins"

but when I try to call for test
the server replay:
"The program can't start because gdal.204.dll is missing from your computer. Try reinstalling program to fix."
I've tried.... never

**Comment by gioman** [comment]
> Thanks for your suggestion, but I suppose, is it incomplete?

@FRANCOSIBILLINI it is complete and working. Do *not* user any LAMP/XAMPP installations. Use the binaries (no installation needed, just unzip) and the config files available in the repo I pointed you to. Install QGIS Server with OSGeo4W as I explain in the tutorial. If you habe previous installations of osgeo4w/apache/etc. I suggest you to remove everything and start from a clean state.

**Comment by FRANCOSIBILLINI** [comment]
Thanks gioman,
![Immagine](https://user-images.githubusercontent.com/37626686/71178705-0ba07200-226f-11ea-9db3-3bf2b148d327.png)

I'VE FOLLOWED YOUR ISTRACTION,
but the installer downloaded from  
http://download.osgeo.org/osgeo4w/osgeo4w-setup-x86_64.exe
has neither apache nor a php.

**Comment by gioman** [comment]
> I'VE FOLLOWED YOUR ISTRACTION,
> but the installer downloaded from
> http://download.osgeo.org/osgeo4w/osgeo4w-setup-x86_64.exe
> has neither apache nor a php.

@FRANCOSIBILLINI no does not. This is why in the repo I pointed you to you find ZIPs with both Apache and PHP pre-configured.

**Comment by FRANCOSIBILLINI** [comment]
Hi many thanks,

i didn't saw the zip file.
I downloaded and made c:\webserver\bla bla
apache works
php works
qgis server goes internal error...

Please let ask:
I've put the file of: OSGeo4W64 this way:
c:\OSGeo4W64\bla bla 
Perhaps I have to put inside c:\webserver\?
c:\webserver\OSGeo4W64\.... ?

**Comment by FRANCOSIBILLINI** [comment]
![Immagine](https://user-images.githubusercontent.com/37626686/71196059-82e5fe00-228f-11ea-9e92-0d27a09d6ed2.png)

Hi Giovanni,

many thanks i didn't seen the zip file...
I've dezipped int c:
C:\webserver
I've used the OSGeo4W64 installer 
I've put (ltr qgis desktop and ltr server anf fcgi + dependencies) into 
C:\OSGeo4W64\
both apache and php works! 
But qgis goes internal server error :(
have a look at picture.
In apache error.log i don't see any error, where i can find the error log inside qgis?
Thanks
